packages/@dcl/ecs/src/systems/videoEvents.ts

```diff
@@ -0,0 +1,79 @@
+import * as components from '../components'
+import { DeepReadonlyObject, Entity, IEngine } from '../engine'
+import { PBVideoEvent } from '../components'
+import { EntityState } from '../engine/entity'
+
+/**
+ * @public
+ */
+export type VideoEventsSystemCallback = (event: DeepReadonlyObject<PBVideoEvent>) => void
+
+/**
+ * @public
+ */
+export interface VideoEventsSystem {
+  removeVideoEventsEntity(entity: Entity): void
+
+  registerVideoEventsEntity(entity: Entity, callback: VideoEventsSystemCallback): void
+}
+
+/**
+ * @internal
+ */
+export function createVideoEventsSystem(engine: IEngine): VideoEventsSystem {
```

Comment:
I think we should prevent default systems as much as possible. This entire system can be replaced by a function to obtain the final state of each video like this:


```suggestion
// this function returns the latest state of the video filtering commands from video events.
// the result also provides some functions to mutate the state of the entity like .play(), .pause() and .seek()
declare function getLatestVideoState(entity: Entity): InternalVideoStateNotComponent
```

We don't need to run this every frame. We may not need events at all if people adopt ECS style of programming like:

```
function mySystem() {
  for (const [entity] of engine.getEntitiesWith(Video)) {
    const video = getLatestVideoState(entity)
    Visible.getMustable(spinnerEntity, { visible: video.currentState != PLAYING })
  }
}
```

Reply:
Uhmmm but the point of these helper systems (also [pointer-events helper system](https://github.com/decentraland/js-sdk-toolchain/blob/8b4e6e0dafaebbfef593428a8e6478512a31c815/packages/%40dcl/ecs/src/systems/events.ts), [raycast helper system](https://github.com/decentraland/js-sdk-toolchain/blob/8b4e6e0dafaebbfef593428a8e6478512a31c815/packages/%40dcl/ecs/src/systems/raycast.ts)) is  that creators can easily just "subscribe" to these specific "events" and not have to implement their own system checking timestamps and stuff.

That's why if we had [the "OnChange" event](https://github.com/decentraland/sdk/issues/740) for components, even these helper systems could be greatly simplifed and let creators have a super friendly way of subscribing to the change of a component (like a pointer event result component, raycast result component or video event component)

Reply:
but those are technical debt also.. @gonpombo8 can go in detail about that.. we used the events because that was the legacy way of doing it to enable the events for react UI, not because we believe it is a good way to expose the state of the components to the scene cretaors

Reply:
the raycast helpers can also be a single function. it inherited the flaw in design intended for React

Reply:
Yes !. We can handle those three events in one single PR just to be consistent.

We can merge this now so we can iterate it after and unblock pravus.

Reply:
cool, now I understand that other approach, I created this issue to tackle the new helper functions as an alternative usage of these "result components": https://github.com/decentraland/sdk/issues/760, feel free to modify it if I miswrote something there